qi-lib/flow/compiler.rkt

```diff
@@ -48,29 +48,27 @@
 
     ;;; Special words
     [((~datum one-of?) v:expr ...)
-     #'(compose
-        ->boolean
-        (curryr member (list v ...)))]
+     #'(qi0->racket (~> (member (list v ...)) ->boolean))]
     [((~datum all) onex:clause)
-     #`(give (curry andmap (qi0->racket onex)))]
+     #`(qi0->racket (~> (>< onex) AND))]
     [((~datum any) onex:clause)
-     #'(give (curry ormap (qi0->racket onex)))]
+     #'(qi0->racket (~> (>< onex) OR))]
     [((~datum none) onex:clause)
      #'(qi0->racket (not (any onex)))]
     [((~datum and) onex:clause ...)
      #'(conjoin (qi0->racket onex) ...)]
     [((~datum or) onex:clause ...)
      #'(disjoin (qi0->racket onex) ...)]
     [((~datum not) onex:clause)
-     #'(negate (qi0->racket onex))]
+     #'(qi0->racket (~> onex NOT))]
     [((~datum gen) ex:expr ...)
      #'(λ _ (values ex ...))]
     [(~or* (~datum NOT) (~datum !))
      #'not]
     [(~or* (~datum AND) (~datum &))
-     #'all?]
+     #'(qi0->racket (>> (and 2> 1>) #t))]
```

Comment:
That was resulting in `#t` as the output instead of the last truthy value, so I swapped it. Also, I had to use opposite folding directions for `AND` and `OR` to get the truthy tests to pass.

Reply:
Ah, indeed. Frankly, even with the note about the fold directions, I find it difficult to remember which is which ;) no fault of yours or Qi's, just my brain.